tremor-pipeline/src/query.rs

```diff
@@ -113,7 +114,19 @@ fn window_decl_to_impl<'script>(
                         )
                     },
                     |interval| {
-                        Ok(TumblingWindowOnTime::from_stmt(interval, ttl, script, stmt).into())
+                        let emit_on_tick = d
+                            .params
+                            .get("emit_on_tick")
+                            .and_then(Value::as_bool)
+                            .unwrap_or_else(|| script.is_none());
```

Comment:
it feels wrong that the default is different based on script being defined, should it perhaps always be false? Or simply forbidden ith scripts?

Reply:
if the script is not defined we definitely use the `ingest_ns`. In that case the tick-time equals the `ingest_ns` time. We can be sure that when a tick exceeded the time for this window, no event will ever come for that window again. This is why i chose `true` for the case of `script.is_none()` here. I hope my assumption is correct.

Reply:
It shouldn't hold off a merge but it's worth thinking if we perhaps just scrap it? I see that if `script.is_none()` we always ignore it (in the other code too) so the only time this has an effect if is `script.is_some()` and it is set to `false`.


My main concern is that this setting has so many interactions with different parts that it'll be hard to explain to operators, I've a hard time wrapping my head around it but the best explanation I came up with is:


> For size-based windows it has no effect. If a script is not set it has no effect. If a script is set and it is set to false it has no effect. If a script is set and it is set to true it will still emit on ticks.

Is that correct? It feels like a very shoot in the footy kind of setting I don't see a lot of use for, might be worth just leaving it out and have the default behavior of `"If you set a script we can't track your time"` to make it easier to gasp?

Reply:
I would have explained it as such in the docs:

```
Windows that use the internal event ingest timestamp as their time source can be tracked and emit with a 100ms resolution if no events arrive. Windows using `script` to configure a custom timestamp cannot be tracked, as the given timestamp might not correlate with the wall-clock time used for the ingest timestamps. If your custom `script` timestamp can be tracked with wall-clock, you can enable the window by setting `emit_on_tick = true`. That makes sense if the timestamp represents a monotonic timestamp that might be off by some constant from the event ingest timestamp.
```

It requires a lot of explanation tbh.
We can totally strip that thing.

Reply:
👍 let's get rid of it!